cpp/include/cudf/datetime.hpp

```diff
@@ -186,6 +204,22 @@ std::unique_ptr<cudf::column> extract_nanosecond_fraction(
   cudf::column_view const& column,
   rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief  Extracts the specified datetime component from any datetime type and
+ * returns an int16_t cudf::column.
+ *
+ * @param column cudf::column_view of the input datetime values
+ * @param component The datetime component to extract
+ * @param mr Device memory resource used to allocate device memory of the returned column
+ *
+ * @returns cudf::column of the extracted int16_t datetime component
+ * @throw cudf::logic_error if input column datatype is not TIMESTAMP
+ */
+std::unique_ptr<cudf::column> extract_datetime_component(
+  cudf::column_view const& column,
+  datetime_component component,
+  rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
```

Comment:
*issue*: We have moved away from raw pointers and now accept `resource_ref`s in these functions:

```suggestion
  rmm::device_async_resource_ref mr = rmm::mr::get_current_device_resource());
```